ui/lesson-renderer/src/main/java/org/cru/godtools/tool/lesson/ui/LessonActivity.kt

```diff
@@ -0,0 +1,44 @@
+package org.cru.godtools.tool.lesson.ui
+
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import org.cru.godtools.base.tool.activity.BaseSingleToolActivity
+import org.cru.godtools.tool.lesson.R
+import org.cru.godtools.tool.lesson.databinding.LessonActivityBinding
+import org.cru.godtools.xml.model.Manifest
+
+@AndroidEntryPoint
+class LessonActivity : BaseSingleToolActivity<LessonActivityBinding>(
+    contentLayoutId = R.layout.lesson_activity,
+    requireTool = true,
+    supportedType = Manifest.Type.LESSON
+) {
+    // region Lifecycle
+    override fun onContentChanged() {
+        super.onContentChanged()
+        binding.setupPages()
+    }
+    // endregion Lifecycle
+
+    // region UI
+    // region Toolbar
+    override val toolbar get() = binding.appbar
+
+    override fun updateToolbarTitle() {
+        title = ""
+    }
+    // endregion Toolbar
+
+    // region Pages
+    @Inject
+    lateinit var lessonPageAdapterFactory: LessonPageAdapter.Factory
+    private val lessonPageAdapter by lazy {
```

Comment:
I looked at the [documentation](https://kotlinlang.org/docs/delegated-properties.html#lazy-properties) for ```by lazy``` but could use some explanation of the value of it being used here. Is it because the factory is not created initially, so we create it the one time, and then that value is remembered for all sequential calls so it doesn't create another factory?

Reply:
there are a few things I want to unpack here.
1. the `by` keyword identifies property delegation. this is a powerful Kotlin feature where you can create property delegates that implement some behavior around setting/getting a property, and all accesses of the property triggers this behavior.
2. the `lazy` delegate defers creation of the property value until it's first use
3. the `lessonPageAdapterFactory` on the previous line is a `lateinit var` and specifically being injected by Dagger once the activity starts.

The reason we need to use `lazy` here is because when the activity object is first created we don't have a value set in `lessonPageAdapterFactory` yet. this means that if we try to create the adapter at object creation it will fail with an NPE. The `lazy` delegate allows us to defer that creation to a latter point after `lessonPageAdapterFactory` has been initialized.